sdk/ml/azure-ai-ml/azure/ai/ml/entities/_compute/_custom_applications.py

```diff
@@ -0,0 +1,220 @@
+            CustomApplicationDefaults.PUBLISHED_PORT,
+        ]
+        min_values = [
+            CustomApplicationDefaults.TARGET_PORT_MIN_VALUE,
+            CustomApplicationDefaults.PUBLISHED_PORT_MIN_VALUE,
+        ]
+        max_values = [
+            CustomApplicationDefaults.TARGET_PORT_MAX_VALUE,
+            CustomApplicationDefaults.PUBLISHED_PORT_MAX_VALUE,
+        ]
+
+        for i, port in enumerate(ports):
+            msg = invalid_value_error_message.format(
+                port_names[i], min_values[i], max_values[i]
+            )
+            if port < min_values[i] or port > max_values[i]:
+                raise ValidationException(
+                    message=msg,
+                    target=ErrorTarget.COMPUTE,
+                    no_personal_data_message=msg,
+                    error_category=ErrorCategory.USER_ERROR,
+                )
+
+    def _to_rest_object(self) -> RestEndpoint:
+        return RestEndpoint(
+            name="connect",
+            target=self.target,
+            published=self.published,
+            protocol=Protocol.HTTP,
+        )
+
+    @classmethod
+    def _from_rest_object(cls, obj: RestEndpoint) -> "EndpointsSettings":
+        return EndpointsSettings(target=obj.target, published=obj.published)
+
+
+class VolumeSettings:
+    """Specifies the Bind Mount settings for a Custom Application.
+
+    :param source: The host path of the mount.
+    :type source: str
+    :param target: The path in the container for the mount.
+    :type target: str
+    """
+
+    def __init__(self, *, source: str, target: str):
+        self.source = source
+        self.target = target
+
+    def _to_rest_object(self) -> RestVolumeDefinition:
+        return RestVolumeDefinition(
+            type=RestVolumeDefinitionType.BIND,
+            read_only=False,
+            source=self.source,
+            target=self.target,
+        )
+
+    @classmethod
+    def _from_rest_object(cls, obj: RestVolumeDefinition) -> "VolumeSettings":
+        return VolumeSettings(source=obj.source, target=obj.target)
+
+
+class CustomApplications:
+    """Specifies the custom service application configuration.
+
+    :param name: Name of the Custom Application.
+    :type name: str
+    :param image: Describes the Image Specifications.
+    :type image: ImageSettings
+    :param endpoints: Configuring the endpoints for the container.
+    :type endpoints: List[EndpointsSettings]
+    :param environment_variables: Environment Variables for the container.
+    :type environment_variables: Optional[Dict[str, str]]
+    :param bind_mounts: Configuration of the bind mounts for the container.
+    :type bind_mounts: Optional[List[VolumeSettings]]
+    """
+
+    def __init__(
+        self,
+        *,
+        name: str,
+        image: ImageSettings,
+        endpoints: List[EndpointsSettings],
+        environment_variables: Optional[Dict] = None,
+        bind_mounts: Optional[List[VolumeSettings]] = None,
```

Comment:
Please add type parameter. Also can we add support for kwargs, and pass it to the properties bag (of Custom Application ) in backend?



Specifically to: https://github.com/Azure/azure-rest-api-specs/blob/6e290d0b310e87d56cfa98cababea00b489d2c07/specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2022-10-01-preview/machineLearningServices.json#L4932

Reply:
This has been done now.